fffw/graph/outputs.py

```diff
@@ -0,0 +1,127 @@
+import os
+from itertools import chain
+from typing import List, Tuple, cast, Optional, Any
+
+from fffw.graph import base
+from fffw.wrapper import BaseWrapper, ensure_binary
+
+__all__ = ['Codec', 'Output', 'OutputList']
+
+
+class Codec(base.Dest, BaseWrapper):
+    index = cast(int, base.Once('index'))
+    """ Index of current codec in ffmpeg output streams."""
+
+    arguments = [
+        ('vbitrate', '-b:v '),
+        ('abitrate', '-b:a '),
+    ]
+
+    def __init__(self, kind: base.StreamType, codec: str,
+                 **kwargs: Any) -> None:
+        super().__init__('', kind)
+        BaseWrapper.__init__(self, **kwargs)
+        self._codec = codec
+
+    @property
+    def map(self) -> Optional[base.InputType]:
+        """ Returns a source or a filter connected to this codec."""
+        if not self.edge:
+            return None
+        return self.edge.input
+
+    @property
+    def name(self) -> str:
+        return f'{self.kind.value}out{self.index}'
+
+    def get_args(self) -> List[bytes]:
+        if self.edge is None:
+            raise RuntimeError("Codec not connected to source")
+        source = self.edge.input
+        if isinstance(source, base.Node):
+            mapping = f'[{self.edge.name}]'
+        else:
+            mapping = source.name
```

Comment:
this if-else decision is made in edge naming function (to take name from source node or from destination), this if shouldn't exist. May be edge names should be automatically wrapped with brackets (`[edge]`)

Reply:
edge makes decision between `[src]` and `[edge]`, and here choices are `src` and `[edge]`. Huge difference.